content/kubermatic/master/tutorials_howtos/applications/create-application-catalogue/_index.en.md

```diff
@@ -0,0 +1,248 @@
+It consists of three parts: metadata on the Application itself, templating method, and the templating source.
+These will be described in more detail in the subsequent paragraphs; For a complete reference, refer to the  [ApplicationDefinition Reference](#applicationdefinition-reference) section.
+Currently ApplicationDefinitions need to be created by hand, but we plan to include a UI importer in an upcoming release.
+
+```yaml
+# Example of an ApplicationDefinition
+apiVersion: apps.kubermatic.k8c.io/v1
+kind: ApplicationDefinition
+metadata:
+  name: prometheus
+spec:
+  description: Prometheus is a monitoring system and time series database.
+  method: helm
+  versions:
+  - template:
+      source:
+        helm:
+          chartName: prometheus
+          chartVersion: 15.10.4
+          url: https://prometheus-community.github.io/helm-charts
+    version: 2.36.2
+  - template:
+      source:
+        git:
+          path: charts/prometheus
+          ref:
+            branch: master
+            remote: https://github.com/prometheus-community/helm-charts
+    version: dev
+```
+
+### Application Metadata
+
+Application Metadata is mainly used for presentation and does not affect how an application is deployed:
+
+- `metadata.name` -> name of the application which will be displayed in the UI
+- `spec.versions[].version` -> Version which is displayed in the UI; Each entry of this field must be unique. We strongly recommend to set this to the version of the Application itself not of the chart.
+
+### Templating Method
+
+Currently the `helm` templating method is supported exclusively.
+
+### Templating Source
+
+Source of the Manifests that should be templated. This controls from where your Application manifests should be fetched. You can combine multiple sources (e.g. helm & git) within one ApplicationDefinition.
+A common use-case for combining is to make stable versions (via helm) and a development version (via git) available at the same time.
+
+#### Helm Source
+
+- `url` ->  URL to a [helm chart repository](https://helm.sh/docs/topics/chart_repository/); If you are unsure about a server, you can always double check if "\<your-url\>/index.yaml" returns a valid index
+- `chartName` -> Name of the chart within the repository
+- `chartVersion` -> Version of the chart; corresponds to the chartVersion field
+
+Currently the best way to obtain `chartName` and `chartVersion` is to make use of `helm search`:
+
+```sh
+# initial preparation
+helm repo add <repo-name> <repo-url>
+helm repo update
+
+# listing the names of all charts in a repository
+helm search repo <repo-name>
+
+# listing versions of a chart
+helm search repo <repo-name>/<chart-name> --versions
+
+# you can also filter for versions. For example if you want to list all prometheus helm charts with a chartversion of 15 or greater, you can run
+helm search repo prometheus-community/prometheus --versions --version ">=15"
+```
+
+#### Git Source
+
```

Comment:
I wonder if we should keep the doc generic for the git source and not mentions the helm chart. it may suggest that git source only works for helm chart which is not a problem for the moment. But if we add a new method, it may be a little confusing. WDYT?

Reply:
I think when we get to the point of adding a new method, we can either go generic or (and I anticipate this more), have the git section split into a "Git Helm" and "Git New Solution" part